bundles/org.openhab.binding.openweathermap/src/main/java/org/openhab/binding/openweathermap/internal/handler/OpenWeatherMapOneCallHandler.java

```diff
@@ -0,0 +1,636 @@
+            }
+            if (forecastDays != newForecastDays) {
+                logger.debug("Rebuilding daily forecast channel groups.");
+                if (forecastDays > newForecastDays) {
+                    if (newForecastDays < 1) {
+                        toBeRemovedChannels.addAll(removeChannelsOfGroup(CHANNEL_GROUP_FORECAST_TODAY));
+                    }
+                    if (newForecastDays < 2) {
+                        toBeRemovedChannels.addAll(removeChannelsOfGroup(CHANNEL_GROUP_FORECAST_TOMORROW));
+                    }
+                    for (int i = newForecastDays; i < forecastDays; ++i) {
+                        toBeRemovedChannels.addAll(
+                                removeChannelsOfGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i)));
+                    }
+                } else {
+                    if (forecastDays == 0 && newForecastDays > 0) {
+                        toBeAddedChannels.addAll(createChannelsForGroup(CHANNEL_GROUP_FORECAST_TODAY,
+                                CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                    if (forecastDays <= 1 && newForecastDays > 1) {
+                        toBeAddedChannels.addAll(createChannelsForGroup(CHANNEL_GROUP_FORECAST_TOMORROW,
+                                CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                    for (int i = (forecastDays < 2) ? 2 : forecastDays; i < newForecastDays; ++i) {
+                        toBeAddedChannels.addAll(
+                                createChannelsForGroup(CHANNEL_GROUP_DAILY_FORECAST_PREFIX + Integer.toString(i),
+                                        CHANNEL_GROUP_TYPE_ONECALL_DAILY_FORECAST));
+                    }
+                }
+                forecastDays = newForecastDays;
+            }
+            ThingBuilder builder = editThing().withoutChannels(toBeRemovedChannels);
+            for (Channel channel : toBeAddedChannels) {
+                builder.withChannel(channel);
+            }
+            updateThing(builder.build());
+        }
+        initialized = true;
+    }
+
+    @Override
+    protected boolean requestData(OpenWeatherMapConnection connection)
+            throws OpenWeatherMapCommunicationException, OpenWeatherMapConfigurationException {
+        logger.debug("Update weather and forecast data of thing '{}'.", getThing().getUID());
+        try {
+            weatherData = connection.getOneCallAPIData(location, forecastMinutes == 0, forecastHours == 0,
+                    forecastDays == 0);
+            return true;
+        } catch (JsonSyntaxException e) {
+            logger.debug("JsonSyntaxException occurred during execution: {}", e.getLocalizedMessage(), e);
+            return false;
+        }
+    }
+
+    @Override
+    protected void updateChannel(ChannelUID channelUID) {
+        // avoid null pointer exceptions and unnecessary work until initiolization is done.
```

Comment:
I got this NullPointerException when I tested the binding in my installation. Seems to be that updateChannel is already called before the initialization was completed. Didn't dig into the details, maybe some synchronization is not accurate in the framework.

Reply:
Wrote a more elaborate comment.

Reply:
Wouldn't it be the better take to add a `null`-check at the place where the NPE happened? Otherwise I would suggest to have a look into the [ThingHandlerHelper.isHandlerInitialized(...)](https://github.com/openhab/openhab-core/blob/1edf25b4f97998d9bcae7936495b3f96ed96a797/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/util/ThingHandlerHelper.java#L25) methods and use framework features to solve your issue.

Reply:
Removed the check to find the NPE, but it did not occur any more. Looks like it was caused by something I changed since then. So finally the check is gone